pypeit/pypeit.py

```diff
@@ -961,34 +960,42 @@ def extract_one(self, frames, det, sciImg, objFind, initial_sky, sobjs_obj):
             slits.mask[flagged_slits] = \
                 slits.bitmask.turn_on(slits.mask[flagged_slits], 'BADSKYSUB')
 
-        msgs.info("Extraction begins for {} on det={}".format(self.basename, det))
```

Comment:
I like the way you restructured this. Indeed if extraction is skipped, there is no need to instantiate the Extract object just to correct the reframe.